src/Control/Functor/Linear/Internal/Class.hs

```diff
@@ -166,3 +208,97 @@ ap f x = f >>= (\f' -> fmap f' x)
 foldM :: forall m a b. Monad m => (b %1 -> a %1 -> m b) -> b %1 -> [a] %1 -> m b
 foldM _ i [] = return i
 foldM f i (x : xs) = f i x >>= \i' -> foldM f i' xs
+
+---------------
+-- Instances --
+---------------
+
+deriving via Generically1 ((,) a)
+  instance Functor ((,) a)
+deriving via Generically1 ((,,) a b)
+  instance Functor ((,,) a b)
+deriving via Generically1 ((,,,) a b c)
+  instance Functor ((,,,) a b c)
+deriving via Generically1 ((,,,,) a b c d)
+  instance Functor ((,,,,) a b c d)
+instance Monoid a => Monad ((,) a) where
+  (a, x) >>= f = go a (f x)
+    where go :: a %1-> (a,b) %1-> (a,b)
+          go b1 (b2, y) = (b1 <> b2, y)
+
+deriving via Generically1 (Sum.Sum f g)
```

Comment:
I don't really like using `Sum.Sum` (because the prefix doesn't give much information about where the type comes from in this case). Maybe you can change `import Data.Monoid.Linear` to `import Data.Monoid.Linear hiding (Sum)`, and then import `Data.Functor.Sum` unqualified? Also, it would be more consistent with your import of `Data.Functor.Compose`

Reply:
`Data.Functor.Compose` only exports one type, and its name doesn't clash with anything else around, so I don't think it's analogous. I could import `Data.Functor.Sum` without renaming though....

Reply:
Well, I don't mind personally, but I think that this is the import style that @aspiwack likes and it's the one that is used (in principle) across the codebase.

Reply:
I just checked, and `Data.Functor.Sum` [only exports one type](https://hackage.haskell.org/package/base-4.16.0.0/docs/Data-Functor-Sum.html). So I think that the situation is pretty analoguous to the `Data.Functor.Compose` or `Data.Functor.Product` case